doc/user/unit-testing.md

```diff
@@ -1,132 +1,261 @@
+
+In this way, you will be able to get the mock object injected into the Reconcile as the following example.
+
+```go
+    route := &routev1.Route{}
+    err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, route)
+    if err != nil {
+        t.Fatalf("get route: (%v)", err)
+    }
+```
+
+**NOTE:** Following an example of issue that can be faced because of an invalid `TypeMeta.APIVersion` informed. It is not recommended declared the `TypeMeta` since it will be implicit generated. 
+
+```shell
+get route: (no kind "Route" is registered for version "v1" in scheme "k8s.io/client-go/kubernetes/scheme/register.go:61")
+```
+
+Following an example which could cause this error. 
+
+```go
+    ...
+    route := &routev1.Route{
+        TypeMeta: v1.TypeMeta{  // TODO (user): Remove the TypeMeta declared
+            APIVersion: "v1",   // the correct value will be `"route.openshift.io/v1"`
+            Kind:       "Route",
+        },
+        ObjectMeta: v1.ObjectMeta{
+            Name:      name,
+            Namespace: namespace,
+            Labels:    ls,
+        },
+    }
+    ...
+```
+Following another example of the issue that can be faced when the third-party resource schema was not added properly.  	
+
+```shell
+create a route: (no kind is registered for the type v1.Route in scheme "k8s.io/client-go/kubernetes/scheme/register.go:61")`
+```
+
+## How to increase the verbosity of the logs?
+
+Following is a snippet code as an example to increase the verbosity of the logs in order to better troubleshoot your tests. 
+
+```go
+import (
+    ...
+    logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+    ...
+)
+func TestMemcachedController(t *testing.T) {	
+    //dev logs
+    logf.SetLogger(logf.ZapLogger(true))
+    ...
+}
+```
 
-[doc_e2e_test]:https://github.com/operator-framework/operator-sdk/blob/2f772d1dc2340dd19bdc3ec8c2dc9f0f77cc8297/doc/test-framework/writing-e2e-tests.md
-[doc_client]:https://github.com/operator-framework/operator-sdk/blob/5c50126e7a112d67826894997eca143e12dc165f/doc/user/client.md
-[doc_cr_fake_client]:https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/fake
-[repo_memcached_reconcile]:https://github.com/operator-framework/operator-sdk-samples/blob/4c6934448684a6953ece4d3d9f3f77494b1c125e/memcached-operator/pkg/controller/memcached/memcached_controller.go#L82
-[doc_reconcile]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler
-[code_test_example]: https://github.com/operator-framework/operator-sdk-samples/blob/master/memcached-operator/pkg/controller/memcached/memcached_controller_test.go#L25
+<!-- Link vars declaration -->
+<!-- NOTE: The CI has a bug and the test will not pass with _ . Use "-" instead of "_" -->
+[doc-e2e-test]:./test-framework/writing-e2e-tests.md
+[doc-client]:./user/client.md
```

Comment:
Can you try doing: 
```
[doc-client]:/doc/user/client.md
```

Reply:
Hi @LiliC,

Yes, I can. However, note that I tried with the full URL as it was before and the same error was faced. Also, I checked and all the lnks are working.

Reply:
Hi @LiliC,

Done. The same issue is faced. Note that I tried to keep the links as it was before as well and it did not work too.

Reply:
@camilamacedo86 This is very weird indeed, so sorry about this! 

@AlexNPavel any ideas why this would be failing? When I check the link manually it works fine.

Reply:
Hi @LiliC,

Yes, it is. My guess that the CI just is checking the files which are changed and is not all docs which have vars for links. In this way, I think we could move forward and create a task to fix the CI.

Reply:
Sorry for all the hassle with this markdown test. The `_` bug combined with the tester following very strict CommonMark formatting that most viewers ignore (just like most browsers ignore incorrect HTML and make assumptions when an error occurs) has been problematic. The issue here was actually very minor. CommonMark requires there to be an empty line before a reference link section (or, more specifically, a link reference definition cannot interrupt a paragraph: https://spec.commonmark.org/0.29/#example-182). However, this document had a comment before the references section, so it was not interpreted as valid reference links. I have pushed the fix for the links (as well as some minor formatting fixes that my editor automatically fixed) to the branch.

Reply:
Really tks @AlexNPavel. I was getting nuts with it :-)